includes/SecurityUtils.php

```
@@ -76,14 +76,30 @@ public static function getSafeBaseUrl(): string
-     * @param string $fragmentDir Absolute path to contents directory.
-     * @param string $rootDir Absolute path to project root directory.
-     * @return array<int, array{slug: string, title: string, mtime: int, filemtime: int}>
+     * @param string $fragmentDir Absolute path to the directory containing content fragments.
+     * @param string $rootDir Absolute path to the project root directory.
+     * @return array<int, array{slug: string, title: string, mtime: int, filemtime: int}> Discovered pages with their slugs, titles, and modification times.
      */
     public static function discoverPages(string $fragmentDir, string $rootDir): array
+    {
+        // Delegate to PerformanceUtils for caching if available
+        if (class_exists('\\CmsForNerd\\PerformanceUtils')) {
+            return PerformanceUtils::getCachedDiscoveredPages($fragmentDir, $rootDir);
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'CmsForNerd\PerformanceUtils' in method 'discoverPages'. (undefined)

(StaticAccess)